src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -131,7 +131,10 @@ private void OnProjectsChanged(ConnectionInformation connection, IEnumerable<Son
             try
             {
                 notifications.ProgressChanged(Strings.ConnectionStepValidatinCredentials);
-                await this.host.SonarQubeService.ConnectAsync(connection, cancellationToken);
+                if (!this.host.SonarQubeService.IsConnected)
```

Comment:
Showing the TE will trigger the connection workflow. If we are already connected, then we want to skip the actual connection step (although we perhaps could have a message saying we are already connected).